packages/mongo/mongo_driver.js

```diff
@@ -1185,9 +1187,16 @@ class AsynchronousCursor {
       return self.fetch();
     } else {
       var results = new LocalCollection._IdMap;
-      await self.forEach(function (doc) {
-        results.set(doc._id, doc);
+
+      const docs = await self._dbCursor.toArray();
+
+      // Need to rewind the cursor otherwise docs will be removed on subsequent polls.
+      this._dbCursor.rewind();
+
+      docs.forEach(doc => {
```

Comment:
`for of` loop will be more efficient:
https://romgrk.com/posts/optimizing-javascript#3-avoid-arrayobject-methods